File: backend/dtos/tuya_device_dto.go
package dtos

// TuyaDeviceDTO represents a single device for API consumers
type TuyaDeviceDTO struct {
	ID          string                `json:"id"`
	Name        string                `json:"name"`
	Category    string                `json:"category"`
	ProductName string                `json:"product_name"`
	Online      bool                  `json:"online"`
	Icon        string                `json:"icon"`
	Status      []TuyaDeviceStatusDTO `json:"status"`
	CustomName  string                `json:"custom_name,omitempty"`
	Model       string                `json:"model,omitempty"`
	IP          string                `json:"ip,omitempty"`
	LocalKey    string                `json:"local_key"`
	GatewayID   string                `json:"gateway_id"`
	CreateTime  int64                 `json:"create_time"`
	UpdateTime  int64                 `json:"update_time"`
}

// TuyaCommandDTO represents a single command
type TuyaCommandDTO struct {
	Code  string      `json:"code" binding:"required"`
	Value interface{} `json:"value" binding:"required"`
}

// TuyaCommandsRequestDTO represents the request body for sending commands
type TuyaCommandsRequestDTO struct {
	Commands []TuyaCommandDTO `json:"commands" binding:"required"`
}

// TuyaIRACCommandDTO represents a single IR AC command request
type TuyaIRACCommandDTO struct {
	InfraredID string `json:"infrared_id" binding:"required"`
	RemoteID   string `json:"remote_id" binding:"required"`
	Code       string `json:"code" binding:"required"`
	Value      int    `json:"value"`
}

// TuyaDeviceStatusDTO represents device status for API consumers
type TuyaDeviceStatusDTO struct {
	Code  string      `json:"code"`
	Value interface{} `json:"value"`
}

// TuyaDevicesResponseDTO represents the response for getting all devices
type TuyaDevicesResponseDTO struct {
	Devices []TuyaDeviceDTO `json:"devices"`
	Total   int             `json:"total"`
}

// TuyaDeviceResponseDTO represents the response for getting a single device
type TuyaDeviceResponseDTO struct {
	Device TuyaDeviceDTO `json:"device"`
}

File: sensio_app/app/src/main/java/com/example/whisper_android/data/remote/dto/TuyaDeviceDto.kt
package com.example.whisper_android.data.remote.dto

import com.google.gson.annotations.SerializedName

data class TuyaDevicesResponseDto(
    @SerializedName("devices") val devices: List<TuyaDeviceDto>,
    @SerializedName("total_devices") val totalDevices: Int,
    @SerializedName("current_page_count") val currentPageCount: Int,
    @SerializedName("page") val page: Int,
    @SerializedName("per_page") val perPage: Int,
    @SerializedName("total") val total: Int
)

data class TuyaDeviceDto(
    @SerializedName("id") val id: String,
    @SerializedName("remote_id") val remoteId: String? = null,
    @SerializedName("name") val name: String,
    @SerializedName("category") val category: String,
    @SerializedName("product_name") val productName: String,
    @SerializedName("online") val online: Boolean,
    @SerializedName("icon") val icon: String,
    @SerializedName("status") val status: List<TuyaDeviceStatusDto>,
    @SerializedName("custom_name") val customName: String? = null
)

data class TuyaDeviceStatusDto(
    @SerializedName("code") val code: String,
    @SerializedName("value") val value: Any
)
